index.d.ts

```diff
@@ -0,0 +1,33 @@
+declare var Viz: VizJs.Viz
+export default Viz
+
+declare namespace VizJs {
+
+    interface Viz {
+        (src: string, opts?: VizOpts): string | HTMLImageElement
+        svgXmlToPngImageElement(svgXml: string, scale: number | undefined, callback: ImageCallback): void
+        svgXmlToPngImageElement(svgXml: string, scale?: number): HTMLImageElement
+        svgXmlToPngBase64(svgXml: string, scale: number | undefined, callback: ImageCallback): void
+    }
+
+    interface ImageCallback {
+        (error: Error | null, image: HTMLImageElement): void
+    }
+
+    type Format = "svg" | "xdot" | "plain" | "ps" | "json" | "png-image-element"
+    type Engine = "circo" | "dot" | "neato" | "osage" | "twopi"
```

Comment:
These could just be strings. I figured since there's a whitelist, might as well encode that, but was also worried about keeping this list up-to-date with the C code. Will back it down to just strings.

I'm unsure how this works with `viz-lite.js`; I'm new to typescript definition files and I think typescript is making some assumptions on project structure that don't quite fit this project:

http://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html

I think typescript is assuming viz.js will add one global variable, but I think viz.js and viz-lite.js both add the same global. I'm going to try to use viz-lite in my typescript project, so we'll see how it goes.